sympy/core/tests/test_assumptions.py

```diff
@@ -726,7 +726,6 @@ def test_Mul_is_infinite():
     assert (x*i).is_finite is None
     assert (f*i).is_finite is False
     assert (x*f*i).is_finite is None
-    assert (z*i).is_finite is False
```

Comment:
This has become redundant as I've added a test which produces `NaN` as the product and I'm sure that `NaN.is_finite is None` is being tested somewhere. So, I think, it's better we just remove it.